### PR TITLE
Add connection management permissions

### PR DESCRIPTION
**Pull request by rmstar**

### CODE REVIEW

File: cdap-ldap-role/src/main/java/io/cdap/cdap/security/authorization/ldap/role/permission/RolePermissionConverter.java [hunk 1]
```diff
@@ -102,12 +102,17 @@ public static List<EntityTypeWithPermission> convertToEntityTypeWithPermission(R
...
+          new EntityTypeWithPermission(EntityType.SYSTEM_APP_ENTITY, StandardPermission.CREATE),
+          new EntityTypeWithPermission(EntityType.SYSTEM_APP_ENTITY, StandardPermission.USE),
+          new EntityTypeWithPermission(EntityType.SYSTEM_APP_ENTITY, StandardPermission.UPDATE)
         );
       case DELETE_PIPELINE:
         return Arrays.asList(
           new EntityTypeWithPermission(EntityType.APPLICATION, StandardPermission.DELETE),
-          new EntityTypeWithPermission(EntityType.DATASET, StandardPermission.DELETE)
+          new EntityTypeWithPermission(EntityType.DATASET, StandardPermission.DELETE),
+          new EntityTypeWithPermission(EntityType.DATASET, StandardPermission.DELETE),
```
Comment (dli357) [inline cdap-ldap-role/src/main/java/io/cdap/cdap/security/authorization/ldap/role/permission/RolePermissionConverter.java]: Duplicate line?

File: cdap-ldap-role/src/main/java/io/cdap/cdap/security/authorization/ldap/role/permission/RolePermissionConverter.java [hunk 2]
```diff
@@ -152,12 +157,15 @@ public static List<EntityTypeWithPermission> convertToEntityTypeWithPermission(R
           new EntityTypeWithPermission(EntityType.ARTIFACT, StandardPermission.LIST),
           new EntityTypeWithPermission(EntityType.ARTIFACT, StandardPermission.UPDATE),
           new EntityTypeWithPermission(EntityType.ARTIFACT, StandardPermission.DELETE),
-          new EntityTypeWithPermission(EntityType.DATASET, StandardPermission.CREATE)
+          new EntityTypeWithPermission(EntityType.DATASET, StandardPermission.CREATE),
+          new EntityTypeWithPermission(EntityType.SYSTEM_APP_ENTITY, StandardPermission.USE),
```
Comment (dli357) [inline cdap-ldap-role/src/main/java/io/cdap/cdap/security/authorization/ldap/role/permission/RolePermissionConverter.java]: Why does deploying artifacts require USE and UPDATE permissions for system app entities?

File: cdap-ldap-role/src/main/java/io/cdap/cdap/security/authorization/ldap/role/permission/RolePermissionConverter.java [hunk 3]
```diff
@@ -102,12 +102,17 @@ public static List<EntityTypeWithPermission> convertToEntityTypeWithPermission(R
           new EntityTypeWithPermission(EntityType.APPLICATION, StandardPermission.CREATE),
           new EntityTypeWithPermission(EntityType.DATASET, StandardPermission.CREATE),
           new EntityTypeWithPermission(EntityType.APPLICATION, StandardPermission.GET),
-          new EntityTypeWithPermission(EntityType.ARTIFACT, StandardPermission.GET)
+          new EntityTypeWithPermission(EntityType.ARTIFACT, StandardPermission.GET),
+          new EntityTypeWithPermission(EntityType.SYSTEM_APP_ENTITY, StandardPermission.CREATE),
```
Comment (dli357) [inline cdap-ldap-role/src/main/java/io/cdap/cdap/security/authorization/ldap/role/permission/RolePermissionConverter.java]: Does creating pipeline need to have create/update permissions on system app entities?

**Review by dli357 — COMMENTED** [review]
Maybe we should create a MANAGE_SYSTEM_APP_ENTITIES RolePermission for the create/update/delete operations, and attach use to CREATE_PIPELINE and USE_WRANGLER, WDYT?

File: cdap-ldap-role/src/main/java/io/cdap/cdap/security/authorization/ldap/role/permission/RolePermissionConverter.java (hunk 3 above)
Comment (rmstar) [reply cdap-ldap-role/src/main/java/io/cdap/cdap/security/authorization/ldap/role/permission/RolePermissionConverter.java]: removed

File: cdap-ldap-role/src/main/java/io/cdap/cdap/security/authorization/ldap/role/permission/RolePermissionConverter.java (hunk 2 above)
Comment (rmstar) [reply cdap-ldap-role/src/main/java/io/cdap/cdap/security/authorization/ldap/role/permission/RolePermissionConverter.java]: fixed

File: cdap-ldap-role/src/main/java/io/cdap/cdap/security/authorization/ldap/role/permission/RolePermissionConverter.java (hunk 1 above)
Comment (rmstar) [reply cdap-ldap-role/src/main/java/io/cdap/cdap/security/authorization/ldap/role/permission/RolePermissionConverter.java]: fixed

**Comment by rmstar** [comment]
> Maybe we should create a MANAGE_SYSTEM_APP_ENTITIES RolePermission for the create/update/delete operations, and attach use to CREATE_PIPELINE and USE_WRANGLER, WDYT?

Done

**Review by dli357 — APPROVED** [review]

**Pull request merged**